client/src/main/java/io/pravega/client/stream/impl/ReaderGroupStateManager.java

```diff
@@ -462,7 +470,25 @@ void checkpoint(String checkpointName, PositionInternal lastPosition) throws Rea
     boolean isCheckpointSilent(String atCheckpoint) {
         return sync.getState().getCheckpointState().isCheckpointSilent(atCheckpoint);
     }
-    
+
+    @SneakyThrows(CheckpointFailedException.class)
+    public void updateTruncationStreamCutIfNeeded() {
+        ReaderGroupState state = sync.getState();
+        // here add check if RG is Subscriber and autoPublishTruncationStreamCut = true
+        if (state.getConfig().getRetentionConfig().equals(ReaderGroupConfig.RetentionConfig.CONSUMPTION_BASED_AT_LAST_CHECKPOINT)
+        && !state.getCheckpointState().isLastCheckpointPublished()) {
+            // we get here only when this is the reader that has completed the lastCheckpoint
+            Optional<Map<Stream, Map<Segment, Long>>> cuts = state.getPositionsForLastCompletedCheckpoint();
+            if (!cuts.isPresent()) {
+                throw new CheckpointFailedException("Could not get positions for last checkpoint.");
+            }
+            cuts.get().entrySet().stream().forEach(entry ->
```

Comment:
Nit: No need of stream(). `cuts.get().entrySet().forEach(..)`

Reply:
fixed